File: test_project/.vscode/SnipHub/snip/hello.py
# Python Hello World
print("Hello, World!")

# 带函数的 Hello World
def say_hello(name):
    return f"Hello, {name}!"

# 带类的 Hello World
class Greeter:
    def __init__(self, name):
        self.name = name
        
    def greet(self):
        return f"Hello, {self.name}!"

# 测试函数和类
print(say_hello("Python"))
greeter = Greeter("Python OOP")
print(greeter.greet())

File: test_project/.vscode/SnipHub/snip/hello.go
package main

import (
	"fmt"
)

// sayHello 返回问候语
func sayHello(name string) string {
	return fmt.Sprintf("Hello, %s!", name)
}

// Greeter 结构体
type Greeter struct {
	Name string
}

// Greet 方法
func (g Greeter) Greet() string {
	return fmt.Sprintf("Hello, %s!", g.Name)
}

func main() {
	// 基本 Hello World
	fmt.Println("Hello, World!")

	// 使用函数
	fmt.Println(sayHello("Go"))

	// 使用结构体和方法
	greeter := Greeter{Name: "Go OOP"}
	fmt.Println(greeter.Greet())

	// 使用切片
	languages := []string{"Go", "Python", "JavaScript", "Rust"}
	fmt.Println("我熟悉的语言:")
	for i, lang := range languages {
		fmt.Printf("%d. %s\n", i+1, lang)
	}
}
